index.html

```diff
@@ -1949,56 +1949,573 @@ <h2>
+        </dd>
+        <dt>
+          <dfn>Recipient</dfn>
+        </dt>
+        <dd>
+          The name of the recipient or contact person at the address.
+        </dd>
+        <dt>
+          <dfn>Phone number</dfn>
+        </dt>
+        <dd>
+          The phone number of the recipient or contact person at the address.
+        </dd>
+      </dl>
+      <section data-dfn-for="PaymentAddress" data-link-for="PaymentAddress">
         <h2>
-          Creating a <code>PaymentAddress</code>
+          <dfn>PaymentAddress</dfn> interface
         </h2>
+        <pre class="idl">
+          [SecureContext, Exposed=(Window)]
+          interface PaymentAddress {
+            [Default] object toJSON();
+            readonly attribute DOMString city;
+            readonly attribute DOMString country;
+            readonly attribute DOMString dependentLocality;
+            readonly attribute DOMString languageCode;
+            readonly attribute DOMString organization;
+            readonly attribute DOMString phone;
+            readonly attribute DOMString postalCode;
+            readonly attribute DOMString recipient;
+            readonly attribute DOMString region;
+            readonly attribute DOMString sortingCode;
+            readonly attribute FrozenArray&lt;DOMString&gt; addressLine;
+          };
+        </pre>
         <p>
-          The steps to <dfn>create a payment address</dfn> from user-provided
-          input are given by the following algorithm.
+          The <a>PaymentAddress</a> interface represents a <a>physical
+          address</a>.
         </p>
-        <ol data-link-for="PaymentAddress">
-          <li>Let <var>address</var> be a new instance of
-          <a>PaymentAddress</a>.
-          </li>
-          <li>Set <var>address</var>.<a>[[\addressLine]]</a> to the result of
-          splitting the user-provided address line into a <a data-cite=
+        <section>
+          <h2>
+            Internal Constructor
+          </h2>
+          <p class="note">
+            This constructor algorithm is used internally by the user agent.
+            It's not available to scripts.
+          </p>
+          <p>
+            The steps to <dfn data-lt=
+            "PaymentAddress.PaymentAddress()">internally construct a
+            <code>PaymentAddress</code></dfn> with <a>AddressInit</a>
+            <var>details</var> are given by the following algorithm:
+          </p>
+          <ol data-link-for="AddressInit">
+            <li>Let <var>address</var> be a new instance of
+            <a>PaymentAddress</a>.Set
+            <var>address</var>.<a>[[\addressLine]]</a>to the empty frozen
+            array, and all other <a data-lt="PaymentAddress slots">internal
+            slots</a>set to the empty string.
+            </li>
+            <li>If <var>details</var> was not passed to the constructor, return
+            <var>address</var> and terminate this algorithm.
```

Comment:
What constructor?

Is details meant to be an optional argument to this algorithm? If so it should be stated ("with an optional AddressInit details")

These days we're not saying "return and terminate this algorithm"; just returning suffices (like in any reasonable programming language).